### DISCUSSION TITLE
deploy to Google Cloud Run and it reloads in 5 minutes

### DISCUSSION THREAD

**Discussion by s2terminal** [zauberzeug/nicegui#1763] [Q&A]
### Question

I have a NiceGUI app deployed to Google Cloud Run. After 5 minutes of displaying the page, the page automatically reloads. When it reloads, all values in the input form that are being edited are lost. Do you have an idea to avoid reloading?

[Cloud Run times out requests in 5 minutes by default](https://cloud.google.com/run/docs/triggering/websockets#client-reconnects), and when Cloud Run times out requests, the WebSocket connection disconnects, and NiceGUI seems to reload when the WebSocket connection disconnects.

I figured out a way. Using long polling seems to extend the request timeout.

```python
nicegui.globals.socket_io_js_transports = ["polling"]
```

Is this method correct? Is there another way? Is it possible to reconnect WebSocket without reloading?

**Comment by rodja** [comment]
Yes, long polling is a feasible solution. It will eat a bit more resources than websocket and is a bit slower in certain conditions but you wont notice it in the grand scope of things. Another way would be to increase the timeout on Cloud Run. But I think it can not be extended for more than an hour.

A few hours ago I created a feature request to not reload the page when connection is lost: https://github.com/zauberzeug/nicegui/discussions/1761. What a coincidence.

**Reply by s2terminal** [reply]
Thank you. I have been investigating this issue for about a week now. Therefore, I was not aware of the pull request.
I look forward to completing the Pull Request.

**Comment by s2terminal** [comment]
I used the latest version of NiceGUI and deployed to Cloud Run. I have not configured anything special, but it seems to be working well.
Thank you very much.

**Comment by CVxTz** [comment]
Edit: Asked my question here -> https://github.com/zauberzeug/nicegui/discussions/2891
